### PR TITLE
add weaver hummingbird ncdump retry in case of WPS remote provider sporadic error

### PR DESCRIPTION
**Pull request by fmigneault**
In some cases, https://github.com/bird-house/hummingbird will answer with the below error in the WPS status.

```xml
<wps:ExecuteResponse xmlns:gml="http://www.opengis.net/gml" xmlns:ows="http://www.opengis.net/ows/1.1" xmlns:wps="http://www.opengis.net/wps/1.0.0" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.opengis.net/wps/1.0.0 http://schemas.opengis.net/wps/1.0.0/wpsExecute_response.xsd" service="WPS" version="1.0.0" xml:lang="en-US" serviceInstance="https://pavics.ouranos.ca:443/wps?service=WPS&amp;request=GetCapabilities" statusLocation="https://pavics.ouranos.ca:443/wpsoutputs/hummingbird/0f9d2e0e-23dc-11ed-939e-0242ac130016.xml">
  <wps:Process wps:processVersion="4.4.1.1">
    <ows:Identifier>ncdump</ows:Identifier>
    <ows:Title>NCDump</ows:Title>
    <ows:Abstract>Run ncdump to retrieve NetCDF header metadata.</ows:Abstract>
  </wps:Process>
  <wps:Status creationTime="2022-08-24T18:39:18Z">
    <wps:ProcessFailed>
      <wps:ExceptionReport>
        <ows:Exception exceptionCode="NoApplicableCode" locater="None">
          <ows:ExceptionText>Process error: Process.py._run_process Line 246 UPDATE statement on table 'pywps_requests' expected to update 1 row(s); -1 were matched.</ows:ExceptionText>
        </ows:Exception>
      </wps:ExceptionReport>
    </wps:ProcessFailed>
  </wps:Status>
</wps:ExecuteResponse>
```

The error seems to be caused by an internal issue with https://github.com/geopython/pywps trying to update some database definition. There is nothing Weaver can do about it. The same process executed multiple times will sometime report that error, and otherwise will succeed completely. 
- example when it worked: https://github.com/bird-house/birdhouse-deploy/pull/259#issuecomment-1226068531
- example when it failed: https://github.com/bird-house/birdhouse-deploy/pull/258#issuecomment-1226028381

In order to avoid failing the test suite sporadically, a retry on failure is attempted up to 10 times with a maximum timeout of 30s.

### CODE REVIEW

File: docs/source/notebook-components/weaver_example.ipynb [hunk 1]
```diff
@@ -1124,7 +1129,11 @@
      "output_type": "stream",
      "text": [
       "Waiting for job completion with pooling monitoring of its status...\n",
-      "Delay: 5s, Duration: 00:00:01, Status: running\n",
+      "Delay: 5s, Duration: 00:00:00, Status: running\n",
+      "Final job status: [failed]\n",
+      "Retrying execution... (1/10)\n",
+      "Job Status Location: [https://pavics.ouranos.ca/weaver/providers/hummingbird/processes/ncdump/jobs/6e23a817-e202-497e-841b-f1d5b2d6eb42] (retry: 1/10)\n",
```
Comment (tlvu) [inline docs/source/notebook-components/weaver_example.ipynb:1135]: nice !

File: docs/source/notebook-components/weaver_example.ipynb [hunk 2]
```diff
@@ -1135,22 +1144,38 @@
...
+    "    if status in [\"failed\", \"succeeded\"]:\n",
+    "        print(f\"Final job status: [{status}]\")\n",
+    "        if status == \"failed\":\n",
+    "            if attempt > 0:\n",
+    "                attempt -= 1\n",
+    "                retry_msg = f\"{retries - attempt}/{retries}\"\n",
+    "                print(f\"Retrying execution... ({retry_msg})\")\n",
+    "                status_location = submit_job()\n",
+    "                print(f\"Job Status Location: [{status_location}] (retry: {retry_msg})\")\n",
+    "                continue\n",
```
Comment (tlvu) [inline docs/source/notebook-components/weaver_example.ipynb:1174]: So this loop will exit if one of timeout or attempt, first one to reach zero?  Just want to ensure no infinite loop.

**Review by tlvu — APPROVED** [review]
LGTM

File: docs/source/notebook-components/weaver_example.ipynb (hunk 2 above)
Comment (fmigneault) [reply docs/source/notebook-components/weaver_example.ipynb:1174]: Yes. First one will break out.

**Pull request merged**